aFWall/src/main/java/dev/ukanth/ufirewall/preferences/SecPreferenceFragment.java

```diff
@@ -258,6 +280,21 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences,
             //passOption = "p" + index;
 
             //currentPosition = index;
+
+            // check if device support fingerprint, if so check if one fingerprint already existed at least
+            if(FingerprintUtil.isAndroidSupport()){
+
+                checkFingerprintDeviceSupport();
+
+            }else{
+
+                if(itemList != null) {
+
+                    itemList.setValueIndex(0);
```

Comment:
I made those lines before the option of fingerprint may be hidden when device not support, so user can be selected but not will not we saved in the configuration.
I will rid of them all.